### DISCUSSION TITLE
What do "project not ready" mean?

### DISCUSSION THREAD

**Discussion by failable** [shuttle-hq/shuttle#860] [Q&A]
Sometimes I get the above error, but after some retries, everything just fine. Then the next time (after some hours maybe) I try to access the service, the above process repeats again, i.e. "project not ready" for the first some times, then OK again.

**Comment by oddgrd** [comment]
When you run `cargo shuttle project start`, a docker container is started for you, and when you run `cargo shuttle deploy` your application is started inside this container. We realize this is a not great DX, and we are working on a refactor of this on a feature branch, where you will only need to run `deploy`, and a container will be started with your application.

The reason you get "project not ready" like this after some time, is likely that your container is going idle, see: https://docs.shuttle.rs/introduction/idle-projects.
